### PR TITLE
RC70: Add Crashpad to Mac

### PR DESCRIPTION
**Pull request by Atlante45**
Adds crash reporting to the mac build.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11231/HighFidelity-Beta-PR13579-107c5d7.apk

**Comment by Atlante45** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11234/HighFidelity-Beta-PR13579-107c5d7.apk

File: cmake/externals/crashpad/CMakeLists.txt [hunk 1]
```diff
@@ -14,21 +12,50 @@ if (WIN32)
       LOG_DOWNLOAD 1
     )
 
+    string(TOUPPER ${EXTERNAL_NAME} EXTERNAL_NAME_UPPER)
```
Comment (birarda) [inline cmake/externals/crashpad/CMakeLists.txt]: Why not keep this at the top since both do this?

File: cmake/modules/MacOSXBundleInfo.plist.in [hunk 2]
```diff
@@ -45,5 +45,25 @@
...
+				<true/>
+				<!--Include to allow HTTP requests-->
+				<key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
+				<true/>
+				<!--Include to specify minimum TLS version-->
+				<key>NSTemporaryExceptionMinimumTLSVersion</key>
+				<string>TLSv1.2</string>
+			</dict>
+		</dict>
+	</dict>
```
Comment (birarda) [inline cmake/modules/MacOSXBundleInfo.plist.in]: Glad we got this to work - any possibility we can get backtrace to update cert on their end so we can remove these changes in the future?

File: interface/src/CrashHandler_Breakpad.cpp [hunk 3]
```diff
@@ -55,11 +57,14 @@ void flushAnnotations() {
...
 }
 
-bool startCrashHandler() {
+bool startCrashHandler(std::string appPath) {
     annotations["version"] = BuildInfo::VERSION;
     annotations["build_number"] = BuildInfo::BUILD_NUMBER;
     annotations["build_type"] = BuildInfo::BUILD_TYPE_STRING;
 
+    auto machineFingerPrint = uuidStringWithoutCurlyBraces(FingerprintUtils::getMachineFingerprint());
+    annotations["machine_fingerprint"] = machineFingerPrint;
```
Comment (birarda) [inline interface/src/CrashHandler_Breakpad.cpp]: We're grabbing this earlier than we were before - have we made sure everything we need to be accessible is ready and we're not delaying app startup with this change?

File: interface/src/workload/GameWorkloadRenderer.cpp [hunk 4]
```diff
@@ -71,7 +70,7 @@ void GameSpaceToRender::run(const workload::WorkloadContextPointer& runContext,
         transaction.resetItem(_spaceRenderItemID, std::make_shared<GameWorkloadRenderItem::Payload>(renderItem));
     }
     
-    transaction.updateItem<GameWorkloadRenderItem>(_spaceRenderItemID, [visible, showProxies, proxies, freezeViews, showViews, views](GameWorkloadRenderItem& item) {
+    transaction.updateItem<GameWorkloadRenderItem>(_spaceRenderItemID, [visible, showProxies, proxies, showViews, views](GameWorkloadRenderItem& item) {
```
Comment (birarda) [inline interface/src/workload/GameWorkloadRenderer.cpp]: Confused by the inclusion of these changes

File: libraries/render-utils/src/Model.cpp [hunk 5]
```diff
@@ -1575,13 +1575,12 @@ void Model::removeMaterial(graphics::MaterialPointer material, const std::string
...
         if (shapeID < _modelMeshRenderItemIDs.size()) {
             auto itemID = _modelMeshRenderItemIDs[shapeID];
-            bool visible = isVisible();
             auto renderItemsKey = _renderItemKeyGlobalFlags;
             bool wireframe = isWireframe();
             auto meshIndex = _modelMeshRenderItemShapes[shapeID].meshIndex;
             bool invalidatePayloadShapeKey = shouldInvalidatePayloadShapeKey(meshIndex);
             bool useDualQuaternionSkinning = _useDualQuaternionSkinning;
-            transaction.updateItem<ModelMeshPartPayload>(itemID, [material, visible, renderItemsKey,
+            transaction.updateItem<ModelMeshPartPayload>(itemID, [material, renderItemsKey,
```
Comment (birarda) [inline libraries/render-utils/src/Model.cpp]: Confused by the inclusion of these changes

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11231/HighFidelity-Beta-PR13579-107c5d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11231/HighFidelity-Beta-Interface-PR13579-107c5d7.dmg

File: interface/src/workload/GameWorkloadRenderer.cpp (hunk 4 above)
Comment (Atlante45) [reply interface/src/workload/GameWorkloadRenderer.cpp]: Just cleaning up some warnings

File: libraries/render-utils/src/Model.cpp (hunk 5 above)
Comment (Atlante45) [reply libraries/render-utils/src/Model.cpp]: Just cleaning up some warnings

File: cmake/externals/crashpad/CMakeLists.txt (hunk 1 above)
Comment (Atlante45) [reply cmake/externals/crashpad/CMakeLists.txt]: Unintended change from moving blocks of code around.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11231/HighFidelity-Beta-PR13579-107c5d7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11231/HighFidelity-Beta-Interface-PR13579-107c5d7.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11234/HighFidelity-Beta-PR13579-107c5d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11234/HighFidelity-Beta-Interface-PR13579-107c5d7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11234/HighFidelity-Beta-PR13579-107c5d7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11234/HighFidelity-Beta-Interface-PR13579-107c5d7.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11236/HighFidelity-Beta-PR13579-b5252da.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11236/HighFidelity-Beta-PR13579-b5252da.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11236/HighFidelity-Beta-Interface-PR13579-b5252da.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11236/HighFidelity-Beta-PR13579-b5252da.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11236/HighFidelity-Beta-Interface-PR13579-b5252da.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11238/HighFidelity-Beta-PR13579-85f56e2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11238/HighFidelity-Beta-PR13579-85f56e2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11238/HighFidelity-Beta-Interface-PR13579-85f56e2.dmg

**Comment by Atlante45** [comment]
## Test Plan:
- Simulate the "Pure Virtual Call" crash on both Mac and Windows and make sure it gets to Backtrace.
- Ensure crash annotations are in the report, especially the machine_fingerprint.

**Comment by QliCknox** [comment]
Pass @conklin94122 

Backtrace had crash logs for both Mac and PC devices

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11253/HighFidelity-Beta-PR13579-f7ead51.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11253/HighFidelity-Beta-PR13579-f7ead51.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11253/HighFidelity-Beta-Interface-PR13579-f7ead51.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11253/HighFidelity-Beta-PR13579-f7ead51.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11253/HighFidelity-Beta-Interface-PR13579-f7ead51.exe

**Pull request merged**